### ISSUE TITLE
Training detail

### ISSUE THREAD

**Issue by YuDamon** [Zengyi-Qin/MonoGRNet#1]
I really appreciate your awesome work. 
I have two questions about training schedule.

1. As your paper said, 

> "We first train the 2D detector, along with the backbone, for 120K iterations using the Adam optimizer . Then the 3D reasoning modules, IDE, 3D localization and local corners, are trained for 80K iterations with the Adam optimizer. Finally, we use SGD to optimize the whole network in an end-to-end fashion for 40K iterations."

For the third training stage for training  "optimize the whole network in an end-to-end fashion",  set the flag 

> joint_2d_3d = True 

or 

> refine= True ?

2. Have you tried other training schedules (like 2D detector --> end2end)? how about the performance?

**Comment by Zengyi-Qin** [comment]
Set joint_2d_3d = True and others = False, in both the decoder/fastBox.py and optimizer/generic_optimizer.py 

As long as you train the 2D first, just feel free to train the 3D branches as you like. We've tried 2D detector --> end2end and it worked well. The most important thing is to train the 2D first. The pretrained 2D model is provided in data/model_2D.pkl and will be automatically loaded. You can disgard the pretrained 3D model by setting "pretrained": false in hypes/kittiBox.json

Thank YOU

**Comment by YuDamon** [comment]
Thanks for reply.
It means the code and pre-trained weights you released have finished the first training stage, is that right? 

I set joint_2d_3d = True and others = False. But if Batch_size = 8, 12GB GPU memory is not enough for training. So I set Batch_size = 4. Is batch_size important to the final performance for this model?

Thanks.

**Comment by Zengyi-Qin** [comment]
The pretrained model (the 3D one) can be directly used for inference. It yields good results on KITTI validation set. The model is not so sensitive to batchsize. I even tried batch_size=1 but it still worked well.
